include/pybind11/detail/common.h

```diff
@@ -476,7 +476,7 @@ template <typename...> struct void_t_impl { using type = void; };
 template <typename... Ts> using void_t = typename void_t_impl<Ts...>::type;
 
 /// Compile-time all/any/none of that check the boolean value of all template types
-#ifdef __cpp_fold_expressions
+#if (!defined(_MSC_VER) || ( _MSC_VER >= 1916 )) && defined(__cpp_fold_expressions)
```

Comment:
Minor nitpick, but can you rewrite this to put the `defined(__cpp_fold_expressions)` conditions first, to make it easier to see what the intention is here?

i.e. "if the compiler has fold expressions and isn't a buggy MS compiler" vs. "if the compiler isn't a buggy MS compiler and has fold expressions"

I haven't been keeping up with MSVC versioning -- does `_MSC_VER == 1915` uniquely identify the buggy release, and are we sure this will be fixed for 15.8.1 Preview 4.1 Late Summer Creator Edition (or whatever MS decides to name the next one)?

Reply:
The developers told me they have the fix "scheduled tentatively for the 15.9 release" which would be compiler version 19.16.